src/ExpressionEvaluator/CSharp/Source/ResultProvider/CSharpFormatter.TypeNames.cs

```diff
@@ -72,7 +79,7 @@ protected override bool AppendSpecialTypeName(StringBuilder builder, Type type,
 
             if (type.IsGenericParameter)
             {
-                AppendIdentifier(builder, escapeKeywordIdentifiers, type.Name);
+                AppendIdentifier(builder, escapeKeywordIdentifiers, type.Name, out sawInvalidIdentifier);
```

Comment:
Why is a type parameter a special type name? And are we testing invalid type parameter names?